src/code/z_lifemeter.c

```diff
@@ -1,17 +1,435 @@
+    sHeartsDDEnv[0][1] = 0;
+    sHeartsDDEnv[0][2] = 0;
+
+    sHeartsDDPrim[1][0] = sHeartsDDPrimColors[ddType][0];
+    sHeartsDDPrim[1][1] = sHeartsDDPrimColors[ddType][1];
+    sHeartsDDPrim[1][2] = sHeartsDDPrimColors[ddType][2];
+
+    sHeartsDDEnv[1][0] = sHeartsDDEnvColors[ddType][0];
+    sHeartsDDEnv[1][1] = sHeartsDDEnvColors[ddType][1];
+    sHeartsDDEnv[1][2] = sHeartsDDEnvColors[ddType][2];
+
+    rFactor = sHeartsDDPrimFactors[ddType][0] * ddFactor;
+    gFactor = sHeartsDDPrimFactors[ddType][1] * ddFactor;
+    bFactor = sHeartsDDPrimFactors[ddType][2] * ddFactor;
+
+    sBeatingHeartsDDPrim[0] = (u8)(rFactor + 255) & 0xFF;
+    sBeatingHeartsDDPrim[1] = (u8)(gFactor + 255) & 0xFF;
+    sBeatingHeartsDDPrim[2] = (u8)(bFactor + 255) & 0xFF;
+
+    rFactor = sHeartsDDEnvFactors[ddType][0] * ddFactor;
+    gFactor = sHeartsDDEnvFactors[ddType][1] * ddFactor;
+    bFactor = sHeartsDDEnvFactors[ddType][2] * ddFactor;
+
+    sBeatingHeartsDDEnv[0] = (u8)(rFactor + 200) & 0xFF;
+    sBeatingHeartsDDEnv[1] = (u8)(gFactor + 0) & 0xFF;
+    sBeatingHeartsDDEnv[2] = (u8)(bFactor + 0) & 0xFF;
+}
+
+s32 func_80100A80(GlobalContext* globalCtx) {
+    gSaveContext.health = (s16)globalCtx->interfaceCtx.unk_24E;
+    
+    return 1;
+}
+
+s32 func_80100AA0(GlobalContext* globalCtx) {
+    InterfaceContext* interfaceCtx = &globalCtx->interfaceCtx;
+
+    interfaceCtx->unk_250 = 0x140;
+    interfaceCtx->unk_24E += 0x10;
+    if (globalCtx->interfaceCtx.unk_24E >= gSaveContext.health) {
+        globalCtx->interfaceCtx.unk_24E = gSaveContext.health;
+        return 1;
+    }
+    return 0;
+}
+
+s32 func_80100AF0(GlobalContext* globalCtx) {
+    InterfaceContext* interfaceCtx = &globalCtx->interfaceCtx;
+    s32 health;
+
+    if (interfaceCtx->unk_250 != 0) {
+        interfaceCtx->unk_250--;
+    } else {
+        interfaceCtx->unk_250 = 0x140;
+        interfaceCtx->unk_24E -= 0x10;
+        if (interfaceCtx->unk_24E <= 0) {
+            interfaceCtx->unk_24E = 0;
+            globalCtx->damagePlayer(globalCtx, -((health = gSaveContext.health) + 1));
+            return 1;
+        }
+    }
+    return 0;
+}
+
+void LifeMeter_Draw(GlobalContext* globalCtx) {
+    s32 pad[5];
+    TexturePtr heartTex;
+    s32 curColorSet;
+    f32 offsetX;
+    f32 offsetY;
+    s32 i;
+    f32 posY;
+    f32 posX;
+    f32 halfTexSize;
+    f32 temp_f4;
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    InterfaceContext* interfaceCtx = &globalCtx->interfaceCtx;
+    Vtx* beatingHeartVtx = interfaceCtx->beatingHeartVtx;
+    s32 fractionHeartCount = gSaveContext.health % 0x10;
+    s16 healthCapacity = gSaveContext.healthCapacity / 0x10;
+    s16 fullHeartCount = gSaveContext.health / 0x10;
+    s32 pad2;
+    f32 lifesize = interfaceCtx->lifeSizeChange * 0.1f;
+    u32 curCombineModeSet = 0;
+    TexturePtr temp = NULL;
+    s32 ddCount = gSaveContext.inventory.dungeonKeys[9] - 1;
+
+    OPEN_DISPS(gfxCtx);
+
+    if (!(gSaveContext.health % 0x10)){
```

Comment:
We usually do this for modulos
```suggestion
    if ((gSaveContext.health % 0x10) != 0){
```